circus-train-aws/src/main/java/com/hotels/bdp/circustrain/aws/AssumeRoleCredentialProvider.java

```diff
@@ -16,45 +16,59 @@
 package com.hotels.bdp.circustrain.aws;
 
 import static com.google.common.base.Preconditions.checkArgument;
-import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.concurrent.TimeUnit;
 
 import org.apache.commons.lang3.StringUtils;
 import org.apache.hadoop.conf.Configuration;
 
 import com.amazonaws.auth.AWSCredentials;
 import com.amazonaws.auth.AWSCredentialsProvider;
 import com.amazonaws.auth.STSAssumeRoleSessionCredentialsProvider;
-import com.amazonaws.auth.STSAssumeRoleSessionCredentialsProvider.Builder;
 
 public class AssumeRoleCredentialProvider implements AWSCredentialsProvider {
 
   public static final String ASSUME_ROLE_PROPERTY_NAME = "com.hotels.bdp.circustrain.aws.AssumeRoleCredentialProvider.assumeRole";
-  private static final int CREDENTIALS_DURATION = 12 * 60 * 60; // max duration in seconds for assumed role credentials
+  public static final String ASSUME_ROLE_CREDENTIAL_DURATION_PROPERTY_NAME = "com.hotels.bdp.circustrain.aws.AssumeRoleCredentialProvider.assumeRoleCredentialDuration";
+  private static final int DEFAULT_CREDENTIALS_DURATION = (int) TimeUnit.HOURS.toSeconds(12); // max duration in seconds for assumed role credentials
 
-  private AWSCredentials credentials;
   private final Configuration conf;
+  private STSAssumeRoleSessionCredentialsProvider credProvider;
 
   public AssumeRoleCredentialProvider(Configuration conf) {
     this.conf = conf;
   }
 
+  private void initializeCredProvider() {
+    String roleArn = conf.get(ASSUME_ROLE_PROPERTY_NAME);
+    int credDuration = conf.getInt(ASSUME_ROLE_CREDENTIAL_DURATION_PROPERTY_NAME, DEFAULT_CREDENTIALS_DURATION);
+
+    checkArgument(StringUtils.isNotEmpty(roleArn),
+        "Role ARN must not be empty, please set: " + ASSUME_ROLE_PROPERTY_NAME);
+
+    // STSAssumeRoleSessionCredentialsProvider should auto refresh its creds in the background.
+    this.credProvider = new STSAssumeRoleSessionCredentialsProvider
+        .Builder(roleArn, "ct-assume-role-session")
+        .withRoleSessionDurationSeconds(credDuration)
```

Comment:
As per @massdosage 
> I'm a bit confused why there are so many other changes to this class when the only change I can really see that is necessary is making this value configurable instead of a hardcoded value?

Reply:
So it's two changes to resolve this issue:
1. Make the assume role time configurable to allow roles that have shorter assume durations than 12 hours
2. Don't cache the credential in our AssumeRoleCredentialProvider. The issue being that we we're holding the token and never refreshing it. If we had a replication that lasted more than 12 hours we'd start failing with expiredToken errors from AWS

Reply:
To expand on that, Ken's changes now enable `STSAssumeRoleSessionCredentialsProvider`'s built-in refresher background thread, so that the user of this class doesn't have to worry about calling refresh manually before the duration times out.

Reply:
OK, thanks for the further explanation, makes sense.